overlord/servicestate/quota_control.go

```diff
@@ -42,6 +50,55 @@ func checkSystemdVersion() {
 
 func init() {
 	checkSystemdVersion()
+	checkMemoryCGroupEnabled()
+}
+
+func checkMemoryCGroupEnabled() {
+	memoryCGroupError = memoryCGroupEnabled()
+}
+
+var CGroupsFilePath = "/proc/cgroups"
+
+//	since the control groups can be enabled/disabled without the kernel config the only
+//	way to identify the status of memory control groups is via /proc/cgroups
+//	"cat /proc/cgroups | grep memory" returns the active status of memory control group
+//	and the 3rd parameter is the status
+//	0 => false => disabled 1 => true => enabled
+func memoryCGroupEnabled() error {
+	cgroupsFile, err := os.Open(CGroupsFilePath)
+	if err != nil {
+		return fmt.Errorf("cannot open cgroups file: %v", err)
+	}
+	defer cgroupsFile.Close()
+	scanner := bufio.NewScanner(cgroupsFile)
+	scanner.Split(bufio.ScanLines)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "memory\t") {
+			memoryCgroupValues := strings.Fields(line)
+			if len(memoryCgroupValues) == 4 { // any change in size should lead to an error for us to update
+				isMemoryEnabled, err := strconv.ParseBool(strings.Fields(line)[3])
+				if err != nil {
+					return fmt.Errorf("cannot parse memory control group status: %v", err)
+				}
+				if !isMemoryEnabled {
+					return fmt.Errorf("memory cgroup disabled on this system")
+				} else {
+					return nil
+				}
+			} else {
+				// change in size, should investigate the new structure
+				return fmt.Errorf("memory control group values doesn't meet size requirements")
```

Comment:
This could be confusing, as most people would take "size" to mean "memory size". Maybe "memory control group configuration couldn't be parsed"?